crates/ntx-builder/src/store.rs

```diff
@@ -22,23 +19,27 @@ use crate::COMPONENT;
 // STORE CLIENT
 // ================================================================================================
 
-type InnerClient = store_client::NtxBuilderClient<InterceptedService<Channel, OtelInterceptor>>;
-
 /// Interface to the store's ntx-builder gRPC API.
 ///
 /// Essentially just a thin wrapper around the generated gRPC client which improves type safety.
 #[derive(Clone, Debug)]
 pub struct StoreClient {
-    inner: InnerClient,
+    inner: StoreNtxBuilderClient,
 }
 
 impl StoreClient {
     /// Creates a new store client with a lazy connection.
     pub fn new(store_url: &Url) -> Self {
-        let channel = tonic::transport::Endpoint::try_from(store_url.to_string())
-            .expect("valid gRPC endpoint URL")
-            .connect_lazy();
-        let store = store_client::NtxBuilderClient::with_interceptor(channel, OtelInterceptor);
+        // SAFETY: The store_url is always valid as it is created from a `Url`.
+        let store = Builder::<WantsTls>::new(store_url.to_string())
+            .expect("Failed to initialize store endpoint")
+            .without_tls()
+            .without_timeout()
+            .without_metadata_version()
+            .without_metadata_genesis()
+            .connect_lazy::<StoreNtxBuilder>()
+            .expect("Failed to build client");
```

Comment:
A superficial comment: `WantsTls` and then `.without_tls()` seems somewhat contradictory.

Reply:
I removed the turbofish and it is just `Builder::new`